src/components/UserSelector.tsx

```diff
@@ -12,24 +26,41 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsClickedOnMenu(prevState => !prevState)}
         >
-          <span>Choose a user</span>
+          <span>{title}</span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
-        <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+      {isClickedOnMenu && (
+        <div className="dropdown-menu" id="dropdown-menu" role="menu">
+          <div className="dropdown-content">
+            {users.map(user => {
+              return (
+                <a
+                  key={user.id}
+                  href={`#user-${user.id}`}
+                  className={classNames(
+                    'dropdown-item', { 'is-active': user.id === selectedUser },
+                  )}
+                  onClick={() => {
+                    onChooseUser(user.id);
+                    setSelectedUser(user.id);
+                    setIsClickedOnMenu(false);
+                    setTitle(user.name);
+                  }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)